thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/CompositePipelineConfigTranslator.java

```diff
@@ -225,9 +226,16 @@ YamlTranslationResult translateYaml() {
       }
     }
     Map<String, Object> dimensionWrapperProperties = buildDimensionWrapperProperties();
-    Map<String, Object> properties = buildWrapperProperties(ChildKeepingMergeWrapper.class.getName(),
-        Collections.singletonList(
-            buildWrapperProperties(DimensionWrapper.class.getName(), nestedPipelines, dimensionWrapperProperties)), this.mergerProperties);
+    Map<String, Object> properties = buildWrapperProperties(
+        ChildKeepingMergeWrapper.class.getName(),
+        Collections.singletonList(buildWrapperProperties(DimensionWrapper.class.getName(), nestedPipelines, dimensionWrapperProperties)),
+        this.mergerProperties);
+
+    List<Map<String, Object>> grouperYamls = getList(yamlConfig.get(PROP_GROUPER));
+    if (!grouperYamls.isEmpty()) {
+      properties = buildGroupWrapperProperties(grouperYamls.get(0), properties);
```

Comment:
So we only apply the first grouper configured in YAML? What about the others.

Reply:
Grouper is the last layer of a metric detection pipeline. At this stage we plan to support only one grouper.

Reply:
If we only want one grouper then should add a validation to only allow one grouper.

Reply:
Added validation